enumerable_method.rb

```diff
@@ -0,0 +1,123 @@
+module Enumerable
+
+    def my_each 
+      i=0
+      while i<self.length
+       yield(self[i])
+       i+=1
+      end
+      self
+    end
+    
+    def my_each_with_index
+      i=0
+      while i<self.length
+        yield(self[i], i )
+        i+=1
+      end
+      self
+    end
+  
+    def my_select 
+      array = []
+        self.my_each do |x|
+          if yield(x)
+            array << x
+          end
+        end
+        array
+    end
+  
+    def my_all? 
+      flag = true
+      self.my_each do |x|
+        if yield(x) == false 
+          flag = false
+        end
+      end
+      flag
+    end
+  
+    def my_any?
+      flag = false
+      self.my_each do |x|
+        flag =  yield(x)
+         break if flag
+      end
+        flag  
+    end
+    
+    def my_none?
+      flag = false
+      self.my_each do |x|
+        flag = !yield(x)
+        break if not flag  
+      end 
+      flag
+    end
+  
+    def my_count 
+      count = 0
+      self.my_each do |x|
+        if block_given?
+        if yield(x) 
+          count += 1
+        
+        else 
+          count += 1
+        end
+      end
+      end
+      count
+    end
+  
+    def my_map(&block) 
+      array = []
+      self.my_each do |x|
+        array.push(yield(x))
+      end
+      array
+  
+    end
+  
+    def my_inject (*item)
+      val = self[0]
+      for i in self[1..self.length] do
+        val=yield(val,i)
```

Comment:
Make sure that code is clean: indentation with two spaces, line lengths, spaces around operators - first section of [this link](https://github.com/github/rubocop-github/blob/master/STYLEGUIDE.md#ruby-style-guide).